src/analysis/processing/qgsalgorithmrasterminmax.cpp

```diff
@@ -0,0 +1,237 @@
+    throw QgsProcessingException( QObject::tr( "Invalid band number for BAND (%1): Valid values for input raster are 1 to %2" ).arg( mBand )
+                                  .arg( layer->bandCount() ) );
+
+  mInterface.reset( layer->dataProvider()->clone() );
+  mHasNoDataValue = layer->dataProvider()->sourceHasNoDataValue( band );
+  mLayerWidth = layer->width();
+  mLayerHeight = layer->height();
+  mExtent = layer->extent();
+  mCrs = layer->crs();
+  mRasterUnitsPerPixelX = std::abs( layer->rasterUnitsPerPixelX() );
+  mRasterUnitsPerPixelY = std::abs( layer->rasterUnitsPerPixelY() );
+  return true;
+}
+
+QVariantMap QgsRasterMinMaxAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  QString dest;
+  std::unique_ptr< QgsFeatureSink > sink;
+  if ( parameters.value( QStringLiteral( "OUTPUT" ) ).isValid() )
+  {
+    QgsFields outFields;
+    outFields.append( QgsField( QStringLiteral( "value" ), QMetaType::Type::Double, QString(), 20, 8 ) );
+    outFields.append( QgsField( QStringLiteral( "extremum" ), QMetaType::Type::QString ) );
+    sink.reset( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, dest, outFields, Qgis::WkbType::Point, mCrs ) );
+    if ( !sink )
+      throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+  }
+
+  const int extractType = parameterAsInt( parameters, QStringLiteral( "EXTRACT" ), context );
+
+  const int maxWidth = QgsRasterIterator::DEFAULT_MAXIMUM_TILE_WIDTH;
+  const int maxHeight = QgsRasterIterator::DEFAULT_MAXIMUM_TILE_HEIGHT;
+  const int numberBlocksWidth = static_cast< int >( std::ceil( 1.0 * mLayerWidth / maxWidth ) );
+  const int numberBlocksHeight = static_cast< int >( std::ceil( 1.0 * mLayerHeight / maxHeight ) );
+  const int numberBlocks = numberBlocksWidth * numberBlocksHeight;
+
+  QgsRasterIterator iter( mInterface.get() );
+  iter.startRasterRead( mBand, mLayerWidth, mLayerHeight, mExtent );
+
+  int iterLeft = 0;
+  int iterTop = 0;
+  int iterCols = 0;
+  int iterRows = 0;
+  std::unique_ptr< QgsRasterBlock > rasterBlock;
+
+  double rasterMinimum = std::numeric_limits< double >::quiet_NaN();
+  double rasterMaximum = std::numeric_limits< double >::quiet_NaN();
+  QgsPointXY rasterMinPoint;
+  QgsPointXY rasterMaxPoint;
+
+  auto blockRowColToXY = [this]( const QgsRectangle & blockExtent, int row, int col ) -> QgsPointXY
+  {
+    return QgsPointXY( blockExtent.xMinimum() + mRasterUnitsPerPixelX * ( col + 0.5 ),
+                       blockExtent.yMaximum() - mRasterUnitsPerPixelY * ( row + 0.5 ) );
+  };
+
+  QgsRectangle blockExtent;
+  while ( iter.readNextRasterPart( mBand, iterCols, iterRows, rasterBlock, iterLeft, iterTop, &blockExtent ) )
+  {
+    feedback->setProgress( 100 * ( ( iterTop / static_cast< double >( maxHeight * numberBlocksWidth ) ) + iterLeft / static_cast< double >( maxWidth ) ) / numberBlocks );
```

Comment:
just a thought: would be nice if the QgsRasterIterator could provide that progress percentage...

Reply:
Ok, done!